### PR TITLE
Endpoint routing in SPA templates

### LINKED ISSUE #9108: SPA templates need to be updated for endpoint routing

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Fixes #9108.

### CODE REVIEW

File: src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/Startup.cs [hunk 1]
```diff
@@ -92,12 +92,11 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
             app.UseAuthentication();
             app.UseIdentityServer();
 #endif
-
-            app.UseMvc(routes =>
+            app.UseEndpoints(endpoints =>
```
Comment (rynowak) [inline src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/Startup.cs]: This should also change to be similar to the other templates in `ConfigureServices`. This should have `AddControllersWithViews()`, and also `AddRazorPages()` when Identity is present.

File: src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/Startup.cs [hunk 2]
```diff
@@ -92,12 +92,11 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
             app.UseAuthentication();
```
Comment (rynowak) [inline src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/Startup.cs]: This should have `UseRouting()` above it, and `UseAuthorization()` after it.

Comment (rynowak) [reply src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/Startup.cs]: With endpoint routing, the auth is handled by middleware now, so the authN/AuthZ middleware sit in between `UseRouting` and `UseEndpoints`

File: src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/Startup.cs [hunk 3]
```diff
@@ -87,17 +99,20 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
...
             app.UseSpaStaticFiles();
-#if (IndividualLocalAuth)
 
+            app.UseRouting();
+
+#if (IndividualLocalAuth)
             app.UseAuthentication();
             app.UseIdentityServer();
 #endif
+            app.UseAuthorization();
```
Comment (danroth27) [inline src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/Startup.cs]: I think authorization should only get included when you aren't doing any authentication.

File: src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/Startup.cs [hunk 4]
```diff
@@ -53,8 +53,20 @@ public void ConfigureServices(IServiceCollection services)
             services.AddAuthentication()
                 .AddIdentityServerJwt();
 #endif
-            services.AddMvc(options => options.EnableEndpointRouting = false);
-
+#if (OrganizationalAuth)
```
Comment (danroth27) [inline src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/Startup.cs]: Do we now support org auth with the SPA templates? I didn't think we did...

File: src/ProjectTemplates/Web.Spa.ProjectTemplates/content/React-CSharp/Startup.cs [hunk 5]
```diff
@@ -53,7 +53,21 @@ public void ConfigureServices(IServiceCollection services)
             services.AddAuthentication()
                 .AddIdentityServerJwt();
 #endif
-            services.AddMvc(options => options.EnableEndpointRouting = false);
+
+#if (OrganizationalAuth)
```
Comment (danroth27) [inline src/ProjectTemplates/Web.Spa.ProjectTemplates/content/React-CSharp/Startup.cs]: Do we now support org auth with the SPA templates? I didn't think we did...

File: src/ProjectTemplates/Web.Spa.ProjectTemplates/content/React-CSharp/Startup.cs [hunk 6]
```diff
@@ -87,17 +101,20 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
...
             app.UseSpaStaticFiles();
-#if (IndividualLocalAuth)
 
+            app.UseRouting();
+
+#if (IndividualLocalAuth)
             app.UseAuthentication();
             app.UseIdentityServer();
 #endif
+            app.UseAuthorization();
```
Comment (danroth27) [inline src/ProjectTemplates/Web.Spa.ProjectTemplates/content/React-CSharp/Startup.cs]: I think `UseAuthorization` should only be included when authentication is enabled.

File: src/ProjectTemplates/Web.Spa.ProjectTemplates/content/ReactRedux-CSharp/Startup.cs [hunk 7]
```diff
@@ -23,7 +23,20 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddMvc(options => options.EnableEndpointRouting = false);
+#if (OrganizationalAuth)
```
Comment (danroth27) [inline src/ProjectTemplates/Web.Spa.ProjectTemplates/content/ReactRedux-CSharp/Startup.cs]: Do we now support org auth with the SPA templates? I didn't think we did...

File: src/ProjectTemplates/Web.Spa.ProjectTemplates/content/ReactRedux-CSharp/Startup.cs [hunk 8]
```diff
@@ -55,11 +68,19 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
...
             app.UseSpaStaticFiles();
 
-            app.UseMvc(routes =>
+            app.UseRouting();
+
+#if (IndividualLocalAuth)
+            app.UseAuthentication();
+            app.UseIdentityServer();
+#endif
+            app.UseAuthorization();
```
Comment (danroth27) [inline src/ProjectTemplates/Web.Spa.ProjectTemplates/content/ReactRedux-CSharp/Startup.cs]: Remove if authentication isn't enabled?

**Review by danroth27 — CHANGES_REQUESTED** [review]

**Comment by ryanbrandenburg** [comment]
:up::date:

**Review by danroth27 — APPROVED** [review]
LGTM

**Comment by Tratcher** [comment]
What's with all the doc deletions?

**Comment by ryanbrandenburg** [comment]
Huh, must have happened when I rebased, will fix.

**Comment by ryanbrandenburg** [comment]
@rynowak better?

**Comment by ryanbrandenburg** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=250430

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact aspnetcore-build@microsoft.com.

I've triaged the above build. I've created/commented on the following issue(s)
https://github.com/aspnet/AspNetCore-Internal/issues/2763

**Review by rynowak — APPROVED** [review]

**Comment by JunTaoLuo** [comment]
FYI you can ignore the AspNetCore-pr-validation-temp configuration. That shouldn't be running for PRs to master.

**Pull request merged**